src/main/java/ch/wisv/areafiftylan/service/TicketServiceImpl.java

```diff
@@ -84,18 +83,22 @@ public TicketTransferToken setupForTransfer(Long ticketId, String goalUserName){
         User u = userService.getUserByUsername(goalUserName).orElseThrow(() -> new UsernameNotFoundException("User " + goalUserName + " not found."));
         Ticket t = ticketRepository.findOne(ticketId);
 
-        TicketTransferToken ttt = new TicketTransferToken(u, t);
+        if (tttRepository.findByTicketId(ticketId).isPresent()) {
```

Comment:
Damn, as I currently see it the tttRepository can only query for one token. It's possible that there are multiple. This has to be fixed. Also this currently only checks the presence of such a token, not if it's valid.